src/main/java/Duke.java

```diff
@@ -1,10 +1,124 @@
+import java.util.Scanner;
+import java.util.Arrays;
+
 public class Duke {
+    private static Task[] taskList = new Task[100];
+    private static int taskSize = 0;
+    public static void printHorizontalLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void printGreeting() {
+        printHorizontalLine();
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        printHorizontalLine();
+    }
+
+    public static void printExitText() {
+        System.out.println("Bye. Hope to see you again soon!");
+        printHorizontalLine();
+    }
+
+    public static void echo(String input) {
+        System.out.println(input);
+        printHorizontalLine();
+    }
+
+    public static void addItem(Task task) {
+        taskSize++;
+        taskList[taskSize] = task;
+    }
+
+    public static void printAddItemText(Task input) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println(input.printTask());
+        System.out.println("Now you have " + taskSize + " tasks in the list.");
+        printHorizontalLine();
+    }
+
+    public static void printTaskList() {
+        if (taskSize == 0) {
+            System.out.println("There is nothing on your list!");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 1; i < taskSize + 1; i++) {
+                System.out.println(i + "." + taskList[i].printTask());
+            }
+        }
+        printHorizontalLine();
+    }
+
+    public static void printMarkTaskText(int taskNumber, String command, boolean isDone) {
+        String previousIcon = taskList[taskNumber].getStatusIcon();
+        if (previousIcon == "X") {
+            if (isDone) {
+                System.out.println("This task has already been marked!");
+            } else {
+                System.out.println("OK, I've marked this task as not done yet:");
+            }
+        } else {
+            if (!isDone) {
+                System.out.println("This task has already been unmarked!");
+            } else {
+                System.out.println("Nice! I've marked this task as done:");
+            }
+        }
```

Comment:
The code can be restructured to make the happy path unindented as much as possible.